From 99e2027f3faea6263226e797bf726fbc356a992f Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Tue, 25 Sep 2018 17:51:39 +0000
Subject: [PATCH 0925/1077] MANTIS 0051880 change: Illumination device should
 generate a new logging service event platform.idl V3.8

added IlluminationEvent
added PlatformIF.logIllumination


git-svn-id: svn://localhost/SelfServiceCommon/trunk@1382 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/idl/Massai/platform.idl            | 19 +++++++++++++++++--
 1 file changed, 17 insertions(+), 2 deletions(-)

diff --git a/SelfServiceCommon/Massai/idl/Massai/platform.idl b/SelfServiceCommon/Massai/idl/Massai/platform.idl
index d1e71c37..7c09e2ad 100644
--- a/SelfServiceCommon/Massai/idl/Massai/platform.idl
+++ b/SelfServiceCommon/Massai/idl/Massai/platform.idl
@@ -27,7 +27,7 @@
 
 module PlatformManagement
 {
-    const string interface_version = "3.7";
+    const string interface_version = "3.8";
 
     /** Data type for text lists. */
     typedef sequence<string> list;
@@ -263,6 +263,19 @@ module PlatformManagement
         ReaderIds       readers;        // Originating devices
     };
     typedef sequence<ReaderOutput> ReaderOutputs;    
+
+    struct IlluminationEvent
+    {
+        string applicationCompany;
+        string applicationType;
+        long red;                    // range 0-255; red component
+        long green;                  // range 0-255; red component
+        long blue;                   // range 0-255; red component
+        long intensity;              // range 0-255; red component
+        long duration;               // time in ms; illumination duration; -1 for unlimited
+        long blinkOnDuration;        // time in ms; for blinking
+        long blinkOffDuration;       // time in ms; for blinking
+    };
         
     enum ApplicationsState
     {
@@ -374,7 +387,9 @@ module PlatformManagement
         code rmvListener(in AppStateListener listener);
 
         void logReaderOutput(in string unitID, in LoggingClass loggingClass, in ReaderOutputs readerData);
-		};
+
+        void logIllumination(in string unitID, in IlluminationEvent illuminationEvent);
+    };
 };
 
 #endif //PLATFORM_IDL
-- 
2.41.0.windows.1

